{
    "data_version": "4.0",
    "data_type": "CVE",
    "data_format": "MITRE",
    "CVE_data_meta": {
        "ID": "CVE-2024-3219",
        "ASSIGNER": "cna@python.org",
        "STATE": "PUBLIC"
    },
    "description": {
        "description_data": [
            {
                "lang": "eng",
                "value": "There is a MEDIUM severity vulnerability affecting CPython.\n\nThe\n \u201csocket\u201d module provides a pure-Python fallback to the \nsocket.socketpair() function for platforms that don\u2019t support AF_UNIX, \nsuch as Windows. This pure-Python implementation uses AF_INET or \nAF_INET6 to create a local connected pair of sockets. The connection \nbetween the two sockets was not verified before passing the two sockets \nback to the user, which leaves the server socket vulnerable to a \nconnection race from a malicious local peer.\n\nPlatforms that support AF_UNIX such as Linux and macOS are not affected by this vulnerability. Versions prior to CPython 3.5 are not affected due to the vulnerable API not being included."
            }
        ]
    },
    "problemtype": {
        "problemtype_data": [
            {
                "description": [
                    {
                        "lang": "eng",
                        "value": "n/a"
                    }
                ]
            }
        ]
    },
    "affects": {
        "vendor": {
            "vendor_data": [
                {
                    "vendor_name": "Python Software Foundation",
                    "product": {
                        "product_data": [
                            {
                                "product_name": "CPython",
                                "version": {
                                    "version_data": [
                                        {
                                            "version_affected": "<=",
                                            "version_name": "0",
                                            "version_value": "3.13.0b4"
                                        }
                                    ]
                                }
                            }
                        ]
                    }
                }
            ]
        }
    },
    "references": {
        "reference_data": [
            {
                "url": "https://github.com/python/cpython/pull/122134",
                "refsource": "MISC",
                "name": "https://github.com/python/cpython/pull/122134"
            },
            {
                "url": "https://github.com/python/cpython/issues/122133",
                "refsource": "MISC",
                "name": "https://github.com/python/cpython/issues/122133"
            },
            {
                "url": "https://mail.python.org/archives/list/security-announce@python.org/thread/WYKDQWIERRE2ICIYMSVRZJO33GSCWU2B/",
                "refsource": "MISC",
                "name": "https://mail.python.org/archives/list/security-announce@python.org/thread/WYKDQWIERRE2ICIYMSVRZJO33GSCWU2B/"
            },
            {
                "url": "http://www.openwall.com/lists/oss-security/2024/07/29/3",
                "refsource": "MISC",
                "name": "http://www.openwall.com/lists/oss-security/2024/07/29/3"
            },
            {
                "url": "https://github.com/python/cpython/commit/06fa244666ec6335a3b9bf2367e31b42b9a89b20",
                "refsource": "MISC",
                "name": "https://github.com/python/cpython/commit/06fa244666ec6335a3b9bf2367e31b42b9a89b20"
            },
            {
                "url": "https://github.com/python/cpython/commit/0b65c8bf5367625673eafb92f85046a1b31259f2",
                "refsource": "MISC",
                "name": "https://github.com/python/cpython/commit/0b65c8bf5367625673eafb92f85046a1b31259f2"
            },
            {
                "url": "https://github.com/python/cpython/commit/220e31adeaaa8436c9ff234cba1398bc49e2bb6c",
                "refsource": "MISC",
                "name": "https://github.com/python/cpython/commit/220e31adeaaa8436c9ff234cba1398bc49e2bb6c"
            },
            {
                "url": "https://github.com/python/cpython/commit/5f90abaa786f994db3907fc31e2ee00ea2cf0929",
                "refsource": "MISC",
                "name": "https://github.com/python/cpython/commit/5f90abaa786f994db3907fc31e2ee00ea2cf0929"
            },
            {
                "url": "https://github.com/python/cpython/commit/b252317956b7fc035bb3774ef6a177e227f9fc54",
                "refsource": "MISC",
                "name": "https://github.com/python/cpython/commit/b252317956b7fc035bb3774ef6a177e227f9fc54"
            }
        ]
    },
    "generator": {
        "engine": "Vulnogram 0.2.0"
    },
    "source": {
        "discovery": "UNKNOWN"
    },
    "credits": [
        {
            "lang": "en",
            "value": "Ellie"
        }
    ]
}